pkg/domain/site.go

```diff
@@ -0,0 +1,182 @@
+package domain
+
+import (
+	"fmt"
+
+	"github.com/skupperproject/skupper/api/types"
+	"github.com/skupperproject/skupper/pkg/qdr"
+	"github.com/skupperproject/skupper/pkg/utils"
+	"github.com/skupperproject/skupper/pkg/version"
+)
+
+type Site interface {
+	GetName() string
+	GetId() string
+	GetVersion() string
+	GetMode() string
+	GetPlatform() string
+	GetCertAuthorities() []types.CertAuthority
+	SetCertAuthorities(cas []types.CertAuthority)
+	GetCredentials() []types.Credential
+	SetCredentials(credentials []types.Credential)
+	GetIngressClasses() []string
+	GetDeployments() []SkupperDeployment
+	SetDeployments(deployments []SkupperDeployment)
+}
+
+type SiteHandler interface {
+	Create(s Site) error
+	Get() (Site, error)
+	Delete() error
+	Update() error
+	RevokeAccess() error
+}
+
+// SiteCommon base implementation of the Site interface
+type SiteCommon struct {
+	Name            string
+	Id              string
+	Version         string
+	Mode            string
+	Platform        string
+	CertAuthorities []types.CertAuthority
+	Credentials     []types.Credential
+	Deployments     []SkupperDeployment
+}
+
+func (s *SiteCommon) GetCertAuthorities() []types.CertAuthority {
+	return s.CertAuthorities
+}
+
+func (s *SiteCommon) SetCertAuthorities(cas []types.CertAuthority) {
+	s.CertAuthorities = cas
+}
+
+func (s *SiteCommon) GetCredentials() []types.Credential {
+	if s.Credentials == nil {
+		s.Credentials = []types.Credential{}
+	}
+	return s.Credentials
+}
+
+func (s *SiteCommon) SetCredentials(credentials []types.Credential) {
+	s.Credentials = credentials
+}
+
+func (s *SiteCommon) GetDeployments() []SkupperDeployment {
+	return s.Deployments
+}
+
+func (s *SiteCommon) SetDeployments(deployments []SkupperDeployment) {
+	s.Deployments = deployments
+}
+
+func (s *SiteCommon) GetName() string {
+	return s.Name
+}
+
+func (s *SiteCommon) GetId() string {
+	return s.Id
+}
+
+func (s *SiteCommon) GetVersion() string {
+	return s.Version
+}
+
+func (s *SiteCommon) GetMode() string {
+	return s.Mode
+}
+
+func (s *SiteCommon) IsEdge() bool {
+	return s.Mode == qdr.ModeEdge
+}
+
+func (s *SiteCommon) ValidateMinimumRequirements() error {
+	reqMsg := func(field string) error {
+		return fmt.Errorf("%s cannot be empty", field)
+	}
+	if s.Name == "" {
+		return reqMsg("name")
+	}
+	if s.Platform == "" {
+		return reqMsg("platform")
+	}
+	if s.Mode == "" {
+		return reqMsg("mode")
+	}
+	if s.Id == "" {
+		s.Id = utils.RandomId(10)
```

Comment:
suggest generating a uuid for uniqueness rather than random

Reply:
Done. I have also modified the default site name to include the hostname.
It was using just the username, which could easily cause issues with the topology.